packages/integrations/src/vue.ts

```diff
@@ -1,103 +1,298 @@
 import { EventProcessor, Hub, Integration } from '@sentry/types';
-import { getGlobalObject, isPlainObject, logger } from '@sentry/utils';
+import { basename, getGlobalObject, logger, timestampWithMs } from '@sentry/utils';
+import { Integrations as APMIntegrations, Span as SpanClass } from '@sentry/apm';
+
+interface IntegrationOptions {
+  Vue: any;
+  /**
+   * When set to false, Sentry will suppress reporting of all props data
+   * from your Vue components for privacy concerns.
+   */
+  attachProps: boolean;
+  /**
+   * When set to true, original Vue's `logError` will be called as well.
+   * https://github.com/vuejs/vue/blob/c2b1cfe9ccd08835f2d99f6ce60f67b4de55187f/src/core/util/error.js#L38-L48
+   */
+  logErrors: boolean;
+  tracing: boolean;
+  tracingOptions: TracingOptions;
+}
+
+interface TracingOptions {
+  track: boolean | Array<string>;
+  timeout: number;
+  hooks: Array<Hook>;
+}
+
+interface ViewModel {
+  [key: string]: any;
+  $root: object;
+  $options: {
+    [key: string]: any;
+    name?: string;
+    propsData?: { [key: string]: any };
+    _componentTag?: string;
+    __file?: string;
+    $_sentryPerfHook?: boolean;
+  };
+  $once: (hook: string, cb: () => void) => void;
+}
 
 /** JSDoc */
 interface Metadata {
   [key: string]: any;
   componentName?: string;
-  propsData?: {
-    [key: string]: any;
-  };
+  propsData?: { [key: string]: any };
   lifecycleHook?: string;
 }
 
+// https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
```

Comment:
Do we require a particular range of Vue versions for this integration to work?
Didn't find it in code nor https://docs.sentry.io/platforms/javascript/vue/.

Reply:
No, anything v2 will work (which is 2016+, so we don't even mention it)